### PR TITLE
fix(inventory): save agent port if needed (service mode)

### PR DESCRIPTION
**Pull request by stonebuzz**
Save agent port, if possible and if it run as service mode.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by g-bougard — APPROVED** [review]

File: src/Inventory/Inventory.php [hunk 1]
```diff
@@ -194,9 +194,10 @@ public function extractMetadata(): array
         }
 
         $this->metadata = [
-            'deviceid' => $this->raw_data->deviceid,
-            'version' => $this->raw_data->version ?? $this->raw_data->content->versionclient ?? null,
-            'itemtype' => $this->raw_data->itemtype ?? 'Computer',
+            'deviceid'  => $this->raw_data->deviceid,
```
Comment (trasher) [inline src/Inventory/Inventory.php:197]: Alignment changes are not needed :)

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 2]
```diff
@@ -95,6 +95,7 @@ private function checkComputer1($computers_id)
             'date_creation' => $computer->fields['date_creation'],
             'is_recursive' => 0,
             'last_inventory_update' => $computer->fields['last_inventory_update'],
+            'last_boot' => null,
```
Comment (trasher) [inline tests/functionnal/Glpi/Inventory/Inventory.php:98]:
```suggestion
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 3]
```diff
@@ -1272,6 +1275,7 @@ public function testUpdateComputer()
...
```
Comment (trasher) [inline tests/functionnal/Glpi/Inventory/Inventory.php:1278]:
```suggestion
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 4]
```diff
@@ -1484,6 +1488,7 @@ public function testUpdateComputer()
...
```
Comment (trasher) [inline tests/functionnal/Glpi/Inventory/Inventory.php:1491]:
```suggestion
```

**Review by trasher — CHANGES_REQUESTED** [review]
Some changes seems coming from another PR. Maybe rebase this one to take care of latest merged changes on inventory

File: src/Inventory/Inventory.php (hunk 1 above)
Comment (stonebuzz) [reply src/Inventory/Inventory.php:197]: done =)

File: tests/functionnal/Glpi/Inventory/Inventory.php (hunk 3 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Inventory.php:1278]: revert

File: tests/functionnal/Glpi/Inventory/Inventory.php (hunk 4 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Inventory.php:1491]: reverted

File: tests/functionnal/Glpi/Inventory/Inventory.php (hunk 2 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Inventory.php]: revert

File: src/Agent.php [hunk 5]
```diff
@@ -329,6 +329,10 @@ public function handleAgent($metadata)
             $input['tag'] = $metadata['tag'];
         }
 
+        if (isset($metadata['port']) && !is_null($metadata['port'])) {
```
Comment (cedric-anne) [inline src/Agent.php:332]: `isset(null)` returns `false`, no need to do the second check.
```suggestion
        if (isset($metadata['port'])) {
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**